storage/src/http/objects/mod.rs

```diff
@@ -155,7 +156,7 @@ pub struct Object {
     /// such a key.
     pub customer_encryption: Option<CustomerEncryption>,
     /// A user-specified timestamp set on an object.
-    pub custom_time: Option<chrono::DateTime<chrono::Utc>>,
+    pub custom_time: Option<OffsetDateTime>,
```

Comment:
Looking at CI, it seems that json deserialization has failed. It seems that the `serde` option is required for OffsetDateTime type fields.

```
#[serde(with = "time::serde::rfc3339")]
```

https://github.com/yoshidan/google-cloud-rust/actions/runs/3668922502/jobs/6202390043#step:6:124

https://stackoverflow.com/questions/72957627/deserialize-rfc-3339-timestamp-with-serde-to-time-rs-offsetdatetime

Reply:
I think I have updated this in all places where OffsetDateTime is serialized/deserialized, but I didn't find a way to run the tests locally for storage. Would you mind re-running them in CI?

Reply:
https://github.com/yoshidan/google-cloud-rust/actions/runs/3674599739/jobs/6213011033

```
thread 'http::storage_client::test::crud_bucket' panicked at 'called `Result::unwrap()` on an `Err` value: ResponseJson(Error("missing field `lockedTime`", line: 20, column: 5))', storage/src/http/storage_client.rs:2409:14
```

Among the new CI errors, I found that the missing field works by assigning `serde(default)`.

```
#[serde(with = "time::serde::rfc3339::option")]
#[serde(default)]
pub locked_time: Option<time::OffsetDateTime>,
```

Errors in `sign.rs` are under investigation.

Reply:
Interesting with the default :thinking: I added it now to that specific place, but maybe I should add it to all places I added a `serde(with=xyz)`?

Reply:
I added it to all `time::serde` fields, otherwise it cannot deserialize when the field is missing (and they are all Option<T>)